src/coreclr/vm/appdomain.hpp

```diff
@@ -2449,14 +2449,21 @@ class SystemDomain : public BaseDomain
         _ASSERTE(m_pGlobalStringLiteralMap);
         return m_pGlobalStringLiteralMap;
     }
-    static FrozenObjectHeapManager* GetFrozenObjectHeapManager()
+    static FrozenObjectHeapManager* GetFrozenObjectHeapManager(bool initialize = true)
```

Comment:
I would look better to have two different methods instead of `initialize` bool arg. It is what we do in other places, e.g.: https://github.com/dotnet/runtime/blob/ac02b66e41041e1cefd355eacaddb45bcb4f1741/src/coreclr/vm/interoplibinterface_comwrappers.cpp#L211-L224

It allows proper contracts. `LIMITED_METHOD_CONTRACT` for the non-throwing version, and `THROWS; MODE_COOPERATIVE (?)` for the throwing version.

Reply:
(Could you please indentation on this contract while you are on it - https://github.com/dotnet/runtime/blob/ac02b66e41041e1cefd355eacaddb45bcb4f1741/src/coreclr/vm/methodtable.cpp#L4225-L4226)